specification/redisenterprise/resource-manager/Microsoft.Cache/stable/2023-07-01/redisenterprise.json

```diff
@@ -0,0 +1,1887 @@
+        "sasUris": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          },
+          "description": "SAS URIs for the target blobs to import from",
+          "x-ms-secret": true
+        }
+      },
+      "required": [
+        "sasUris"
+      ],
+      "description": "Parameters for a Redis Enterprise import operation."
+    },
+    "ExportClusterParameters": {
+      "title": "Export an RDB file into a target database",
+      "type": "object",
+      "properties": {
+        "sasUri": {
+          "type": "string",
+          "description": "SAS URI for the target directory to export to",
+          "x-ms-secret": true
+        }
+      },
+      "required": [
+        "sasUri"
+      ],
+      "description": "Parameters for a Redis Enterprise export operation."
+    },
+    "OperationStatus": {
+      "description": "The status of a long-running operation.",
+      "type": "object",
+      "properties": {
+        "id": {
+          "type": "string",
+          "description": "The operation's unique id."
+        },
+        "name": {
+          "type": "string",
+          "description": "The operation's name."
+        },
+        "startTime": {
+          "type": "string",
+          "description": "The start time of the operation."
+        },
+        "endTime": {
+          "type": "string",
+          "description": "The end time of the operation."
+        },
+        "status": {
+          "type": "string",
+          "description": "The current status of the operation."
+        },
+        "error": {
+          "$ref": "../../../../../common-types/resource-management/v2/types.json#/definitions/ErrorResponse",
+          "description": "Error response describing why the operation failed."
+        }
+      }
+    },
+    "ForceUnlinkParameters": {
+      "title": "Forcibly unlink another database from this database.",
+      "type": "object",
+      "properties": {
+        "ids": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          },
+          "description": "The resource IDs of the database resources to be unlinked."
+        }
+      },
+      "required": [
+        "ids"
+      ],
+      "description": "Parameters for a Redis Enterprise Active Geo Replication Force Unlink operation."
+    },
+    "FlushParameters": {
+      "title": "Flush all the keys from the current and all its linked databases.",
+      "type": "object",
+      "properties": {
+        "ids": {
+          "type": "array",
+          "items": {
+            "type": "string",
+            "format": "arm-id",
+            "x-ms-arm-id-details": {
+              "allowedResources": [
+                {
+                  "type": "Microsoft.Cache/redisEnterprise/databases"
+                }
+              ]
+            }
+          },
+          "description": "The identifiers of all the other database resources in the georeplication group to be flushed."
+        }
```

Comment:
How would user know which other databases are linked to this database (on which Post action is being performed)? Basically, I mean, is this linking not being handled by service internally, without client passing these as parameters, and service knowing internally the list of linked databases?

Reply:
The service internally knows the list of linked databases and performs on this

The reason we want this information to be provided by the customer is to ensure that the user has the required access on the linked databases as well (as this operation deletes the data on these as well) along with the current database

The user can get this information from the get call on this database.

Reply:
I think you need to use linkedAccessChecks: [https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderRegistration.html#linkedaccesschecks-optional](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderRegistration.html#linkedaccesschecks-optional) for this purpose, where you will callout in your manifest that when flush is being called for a databases, a user is also checked if they have access to the linked databases.

Reply:
Yes, That is already being done!